core/src/main/java/org/eclipse/hono/util/TenantObject.java

```diff
@@ -336,43 +399,63 @@ public TenantObject setAdapterConfigurations(final List<Map<String, Object>> con
      *                       configured adapters or {@code null} in order to
      *                       remove any existing configuration.
      * @return This tenant for command chaining.
+     * @throws IllegalArgumentException if the given configurations does not contain a <em>type</em> or
+     *                                  if more than one of the adapter configurations have the same <em>type</em> or
+     *                                  if error parsing the given configuration.
+     * @deprecated Use {@link #setAdapters(List adapters)} instead.
      */
+    @Deprecated
     @JsonIgnore
     public TenantObject setAdapterConfigurations(final JsonArray configurations) {
-        if (configurations == null) {
-            this.adapterConfigurations = null;
-        } else {
-            this.adapterConfigurations = new LinkedList<>();
-            configurations.stream().filter(obj -> JsonObject.class.isInstance(obj)).forEach(config -> {
-                addAdapterConfiguration((JsonObject) config);
-            });
+        final List<Adapter> adapters = Optional.ofNullable(configurations)
+                .map(ok -> configurations.stream()
+                        .map(JsonObject::mapFrom)
```

Comment:
IMHO at this place we first need to filter for JsonObject instances in order to get the same behavior as in the original implementation, right?

Reply:
I have removed it as it doesn't makes sense to filter out only instances of the `JsonObject` but in the other method `setAdapterConfigurations(final List<Map<String, Object>> configurations)` we don't do it. I agree with you that the behavior should remain the same as in the original implementation. I have again added this filter for _JsonObject_.